tests/conftest.py

```diff
@@ -47,6 +49,11 @@ def spark() -> SparkSession:
         .config("spark.worker.ui.retainedDrivers", "1")
         # fixed memory
         .config("spark.driver.memory", "2g")
+        .config("spark.sql.warehouse.dir", tmp_path_factory.mktemp("warehouse"))
+        .config(
+            "spark.driver.extraJavaOptions",
+            "-Dderby.system.home={tmp_path_factory.mktemp('derby')}",
```

Comment:
My understanding of this is that we are explicitly configuring Dderby as the metadata store for Spark. And by defining specific paths for Spark to write the metadata, we are avoiding clashes between the multiple Spark Sessions created by pytest-xdist.